### DISCUSSION TITLE
Arm 64 support

### DISCUSSION THREAD

**Discussion by Engg-blogger** [bpftrace/bpftrace#3420] [General]
Hi Team - Wanted to check if the current appimage also works for arm 64 processors? If not, do you have guidelines on how to build one for arm 64?

**Comment by viktormalik** [comment]
AFAIK, we currently build the AppImage for x86 only.

I was thinking about trying to cross-compile the AppImage for other arches, too, but didn't get a chance to try that.

**Comment by Engg-blogger** [comment]
Is there a timeline for this? If not, can you refer me to the steps to do this?

**Reply by ajor** [reply]
This is how we build our release AppImage: https://github.com/bpftrace/bpftrace/blob/b2e255870ba010d4a7e4852bffcf1c567b016fd0/scripts/create-assets.sh#L55

More Nix docs:
https://github.com/bpftrace/bpftrace/blob/master/docs/nix.md#build-bpftrace-as-a-statically-linked-binary

I assume you'd just have to run this on an ARM 64 machine to get an ARM AppImage. We haven't got any kind of cross-compilation infrastructure in place right now.

**Reply by Engg-blogger** [reply]
@ajor I tried running all the nix build commands in arm 64 machine. The appimage was created. But when I try to run, I am getting "exec format error". Any suggestions?

**Reply by ajor** [reply]
Sounds like it wasn't compiled with the right architecture. I'd start by checking what architecture your executable was actually built with, but as for how to change it, I'm not sure sorry. Might need some digging through either the Nix or the AppImage official docs.

**Reply by Engg-blogger** [reply]
Thanks @ajor The container image was getting replaced before firing build commands. It worked after i fixed that issue.
